### ISSUE TITLE
Is there a way to list all experiments defined in Matomo?

### ISSUE THREAD

**Issue by tobihille** [innocraft/php-experiments#10]
My Use-Case is that the page is behind a Varnish cache and I need to modify the Vary-header. To make this happen I would like to have a call that works e.g. like this 
`$experiment = new Experiment();`
`$experiment->getExperimentVariation();`

For the time being I will add a config and execute new Experiment(...) for every configured experiment but this is in my opinion a massive overhead.

Thank you 🙂

**Comment by tsteur** [comment]
There is an API method `AbTesting.getActiveExperiments` and `AbTesting.getAllExperiments` see https://developer.matomo.org/api-reference/reporting-api#AbTesting but it's not possible to get this information through this library. It would potentially slow a regular request down quite a bit to get this information on demand. Of course it may be less of an issue if it's cached afterwards in Varnish.

Not sure if this answers your question?

**Comment by tobihille** [comment]
Ah I see. And I guess that's also the reason why e.g. the percentage of experiments need to be defined via code and not fetched from the matomo instance.

In fact it still does matter a lot in my use-case because not every page view is a cache-hit and I aim for a fast result on cache-misses, too.

I'll check the referenced API (btw: thank you a lot for mentioning it) but I developed a solution that works at least for now without communication to Matomo (by creating a single point of entry in our codebase and then collecting all variants that a user is passed though to set the correct header).

In case somebody will read this in the future or you are just curious: In essence I only implemented "Step 3" from this blog post: https://info.varnish-software.com/blog/live-ab-testing-varnish-and-vcs by using this lib and I'm ready to test it after just one day. So thank you for this great library!
